src/script/DeployBase.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// slither-disable-start reentrancy-benign
+
+pragma solidity ^0.8.23;
+
+import {Script} from "forge-std/Script.sol";
+import {IEarningPowerCalculator} from "../interfaces/IEarningPowerCalculator.sol";
+import {Staker} from "../Staker.sol";
+
+abstract contract DeployBase is Script {
```

Comment:
I think some top level natspec for the various deploy scripts added in this PR are warranted, given their intended use by third part developers. This would include `@title` and `@author` tags, along with, of course `@notice` tags that go into some detail about how the various scripts are intended to be used by third party devs.

Reply:
https://github.com/withtally/staker/pull/126/commits/e8c1a223d9fd3bad357550dc662996a86f12828c